pyogrio/geopandas.py

```diff
@@ -19,6 +29,26 @@ def _stringify_path(path):
     return path
 
 
+def _try_parse_datetime(ser):
+    import pandas as pd  # only called when pandas is known to be installed
+
+    if PANDAS_GE_20:
+        datetime_kwargs = dict(format="ISO8601", errors="ignore")
+    else:
+        datetime_kwargs = dict(yearfirst=True)
+    res = pd.to_datetime(ser, **datetime_kwargs)
```

Comment:
This first attempt will already raise a warning with the latest pandas, so we will need to catch that warning:

> FutureWarning: In a future version of pandas, parsing datetimes with mixed time zones will raise a warning unless `utc=True`. Please specify `utc=True` to opt in to the new behaviour and silence this warning. To create a `Series` with mixed offsets and `object` dtype, please use `apply` and `datetime.datetime.strptime`

(I have to check if the warning is actually wrong about the future behaviour of raising a warning. I would assume the deprecation would warn for an error in the future. In which case we should also already start catching that, I think)